core/src/main/scala/kafka/utils/CommandLineUtils.scala

```diff
@@ -64,7 +64,7 @@ object CommandLineUtils extends Logging {
 
     val props = new Properties
     for(a <- splits) {
-      if (a.length == 1) props.put(a(0), "")
+      if (a.length == 1) throw new IllegalArgumentException(s"Missing value for key ${a(0)}")
```

Comment:
Another option is to look for keys with empty values after this method returns. I'd be interested in what others think too.